lib/connection/connection.js

```diff
@@ -411,6 +421,164 @@ function Connection(context)
     return this;
   };
 
+  /**
+   * Gets the response containing the status of the query based on queryId.
+   *
+   * @param {String} queryId
+   *
+   * @returns {Object} the query response
+   */
+  async function getQueryResponse(queryId) {
+    // Check if queryId exists and is valid uuid
+    Errors.checkArgumentExists(Util.exists(queryId),
+      ErrorCodes.ERR_CONN_FETCH_RESULT_MISSING_QUERY_ID);
+    Errors.checkArgumentValid(queryIdRegex.test(queryId),
+      ErrorCodes.ERR_GET_RESPONSE_QUERY_INVALID_UUID, queryId);
+
+    // Form the request options
+    const options =
+    {
+      method: 'GET',
+      url: Url.format(
+        {
+          pathname: `/monitoring/queries/${queryId}`
+        }),
+    };
+
+    // Get the response containing the query status
+    const response = await services.sf.requestAsync(options);
+
+    return response['data'];
+  }
+
+  /**
+   * Extracts the status of the query from the query response.
+   *
+   * @param {Object} queryResponse
+   *
+   * @returns {String} the query status.
+   */
+  function extractQueryStatus(queryResponse) {
+    const queries = queryResponse['data']['queries'];
+    let status = QueryStatus.code.NO_DATA; // default status
+    if (queries.length > 0) {
+      status = queries[0]['status'];
+    }
+
+    return status;
+  }
+
+  /**
+   * Gets the status of the query based on queryId.
+   *
+   * @param {String} queryId
+   *
+   * @returns {String} the query status.
+   */
+  this.getQueryStatus = async function (queryId) {
+    return extractQueryStatus(await getQueryResponse(queryId));
+  };
+
+  /**
+   * Gets the status of the query based on queryId and throws if there's an error.
+   *
+   * @param {String} queryId
+   *
+   * @returns {String} the query status.
+   */
+  this.getQueryStatusThrowIfError = async function (queryId) {
+    const response = await getQueryResponse(queryId);
+    const queries = response['data']['queries'];
+    const status = extractQueryStatus(response);
```

Comment:
Ok If we can stay with throwing the error in this method lets replace those 3 lines with getQueryStatus method

Reply:
Sure, we can do that. Now the response is only retrieved if info on the error is needed